trajectory_tracker/src/trajectory_tracker.cpp

```diff
@@ -443,9 +458,11 @@ void TrackerNode::control()
   tracking.pose.orientation = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0.0, 0.0, 1.0), -angle));
   pub_tracking_.publish(tracking);
 
-  path_step_done_ = i_nearest - 2;  // i_nearest is the next of the nearest node
+  path_step_done_ = i_nearest - 1;  // i_nearest is the next of the nearest node
```

Comment:
next goal and nearest point are searched from `path_step_done_`. `- 1` is enough to search from current nearest point.